[Android/ARM64, v5] Support ndk-build to run nnstreamer on Android

leemgs opened the pull request:
This commit is to support the ndk-build command that is included in Android
NDK toolkit. This commit is to be enabled in order to run nnstream-based
their applications on Android platform as well as Ubuntu and Tizen.

Signed-off-by: Geunsik Lim <geunsik.lim@samsung.com>
Signed-off-by: Jijoong Moon <Jijoong.moon@amsung.com>

myungjoo commented:
TODO
1. Android.mk should import VERSION value from somewhere and set for nnstreamer.c
2. Do something on the hardcoded arch (armv8)

leemgs commented:
**Changes**

* Version 3
1. Diconnected dependant absolute path of libz.so file
   - It means that nnstreamer library and test app (shared ELF) do not call
     libz.so. file directly. The libz.so file will be called gstreamer to avoid
     absolute path issue as soon as possible.
2. Added multiple architeecture structure
   - This commit support others (e.g. ARM32, X86, X86_64) as well as ARM64.

* Version 2
1. Removed hardcoded operations to support ARM64
2. Added instruction manual to reproduce consistently
3. Synchronized Android NDK + Android Platform (Nougat) + Android API level (24)
4. Unified separated two folders (e.g., header files and .so files)

* Version 1
1. Added Android.mk file to build nnstreamer and apps
2. Added Application.mk to define a run-time C library and
   the API level of Android

leemgs commented:
**Changes**

* Version 4
1. Added variable NNSTREAMER_VERSION to do not modify the existing nnstreamer.c

* Version 3
1. Disconnected dependant absolute path of libz.so file
   - It means that nnstreamer library and test app (shared ELF) do not call
     libz.so. file directly. The libz.so file will be called gstreamer to avoid
     absolute path issue as soon as possible.
2. Added multiple architecture structure
   - This commit support others (e.g. ARM32, X86, X86_64) as well as ARM64.

* Version 2
1. Removed hardcoded operations to support ARM64
2. Added instruction manual to reproduce consistently
3. Synchronized Android NDK + Android Platform (Nougat) + Android API level (24)
4. Unified separated two folders (e.g., header files and .so files)

* Version 1
1. Added Android.mk file to build nnstreamer and apps
2. Added Application.mk to define a run-time C library and
   the API level of Android

leemgs commented:
**CHANGELOG**

* Version 5
1. Added the tensor decoder in Android.mk to avoid a situation that developers
   have to modify the gst_nnstreamer_init() function.

* Version 4
1. Added variable NNSTREAMER_VERSION to do not modify the existing nnstreamer.c

* Version 3
1. Disconnected the dependant absolute path of libz.so file
   - It means that nnstreamer library and test app (shared ELF) do not call
     libz.so. file directly. The libz.so file will be called gstreamer to avoid
     absolute path issue as soon as possible.
2. Added a multiple architecture structure
   - This commit support others (e.g. ARM32, X86, X86_64) as well as ARM64.

* Version 2
1. Removed hardcoded operations to support ARM64
2. Added instruction manual to reproduce consistently
3. Synchronized Android NDK + Android Platform (Nougat) + Android API level (24)
4. Unified separated two folders (e.g., header files and .so files)

* Version 1
1. Added Android.mk file to build nnstreamer and apps
2. Added Application.mk to define a run-time C library and
   the API level of Android

myungjoo commented:
CI / Build test required soon.
